crypto/pkcs11key/key.go

```diff
@@ -30,6 +30,24 @@ var hashPrefixes = map[crypto.Hash][]byte{
 	crypto.RIPEMD160: {0x30, 0x20, 0x30, 0x08, 0x06, 0x06, 0x28, 0xcf, 0x06, 0x03, 0x00, 0x31, 0x04, 0x14},
 }
 
+// Ctx defines the subset of pkcs11.Ctx's methods that we use, so we can inject
+// a different Ctx for testing.
+type Ctx interface {
```

Comment:
By "you can do whatever", I meant that my qualms were at like 60% and I'm not super worried about it.